examples/data-transfer-pg/main.py

```diff
@@ -0,0 +1,91 @@
+''' Main module. '''
+import sys
+import os
+import time
+import pyodbc
+
+CONNECTION_STRING = 'DRIVER={{PostgreSQL Unicode}};SERVER={server};DATABASE={database};UID={username};PWD={password};'
+
+
+def main():
+    ''' App entrypoint. '''
+    # Wait for postgres database server to fully spawn.
+    time.sleep(5)
+
+    db1_name = os.environ['DB1_NAME']
+    db1_conn = get_connection(
+            os.environ['DB1_HOST'], 
+            db1_name, 
+            os.environ['DB1_USERNAME'], 
+            os.environ['DB1_PASSWORD']
+        )
+    db1_cur = db1_conn.cursor()
+
+    db2_name = os.environ['DB2_NAME']
+    db2_conn = get_connection(
+            os.environ['DB2_HOST'], 
+            db2_name,
+            os.environ['DB2_USERNAME'], 
+            os.environ['DB2_PASSWORD']
+        )
+    db2_cur = db2_conn.cursor()
+
+    print(f'Creating fruits table and populating data in {db1_name}.')
+    db1_cur.execute(extract_sql('sql/db1_setup.sql'))
+    db1_cur.execute('INSERT INTO fruits VALUES (1, ?, ?)', ('Banana', 150))
+    db1_cur.execute('INSERT INTO fruits VALUES (2, ?, ?)', ('Orange', 64))
+    db1_cur.execute('INSERT INTO fruits VALUES (3, ?, ?)', ('Apple', 35))
+    db1_conn.commit()
+
+    print(f'Create fruits table in {db2_name} database.')
+    db2_cur.execute(extract_sql('sql/db2_setup.sql'))
+    db2_conn.commit()
+
+    print(f'Extracting fruits data from {db1_name} database.')
+    db1_cur.execute('SELECT * FROM fruits')
+    rows = db1_cur.fetchall()
+
+    print(f'Transferring fruits data to {db2_name} database.')
+    for row in rows:
+        db2_cur.execute('INSERT INTO fruits VALUES (?, ?, ?)', (row.id, row.name, row.quantity))
+    db2_conn.commit()
```

Comment:
Extract this logic to transfer data to a separate function `transfer_data(source_db, dest_db)`.

Reply:
This function requires cursors of both databases and connection of destination database, is this fine or sending just cursors as parameters seem more appropriate?